src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -307,6 +307,34 @@ private static bool IsDomainEqualToHost(string domain, string host)
                    (string.Compare(host, 0, domain, 1, host.Length, StringComparison.OrdinalIgnoreCase) == 0);
         }
 
+        internal static int CompareDomain(string? left, string? right)
+        {
+            if (left is null)
+            {
+                return right is null ? 0 : -1;
+            }
+
+            if (right is null)
+            {
+                return 1;
+            }
+
+            int indexLeft = GetComparisonStartIndex(left);
+            int indexRight = GetComparisonStartIndex(right);
+
+            int result;
+
+            if ((result = (left.Length - indexLeft).CompareTo(right.Length - indexRight)) != 0)
```

Comment:
This but this form seems more readable to me:

```suggestion
            int result = (left.Length - indexLeft).CompareTo(right.Length - indexRight);
            if (result != 0)
```

Reply:
Replaced by bool